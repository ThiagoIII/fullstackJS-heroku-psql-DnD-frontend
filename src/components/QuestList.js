import React from 'react'

const QuestList = ({userInfo}) => {
	return 	<>
				<h3>Quests</h3>
				<ul>
					{userInfo.length > 0  
							? userInfo.map(quest =><>
														<li>{quest.questname}</li>
														<li>{quest.quest}</li>
													</>) 
							: <h4>No quests registered from you!</h4>}
				</ul>
			</>
}

export default QuestList
