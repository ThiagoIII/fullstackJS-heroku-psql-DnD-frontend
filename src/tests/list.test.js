import React from "react";
import renderer from "react-test-renderer";
import QuestList from "../components/QuestList";

const questsList = [
	{
    questname: "quest1",
    quest: "this is the first quest",
	},
	{
    questname: "quest2",
    quest: "this is the second quest",
	},
	{
    questname: "quest3",
    quest: "this is the third quest",
	},
];

it("renders correctly", () => {
	const tree = renderer.create(<QuestList userInfo={questsList} />).toJSON();
	expect(tree).toMatchSnapshot();
});
