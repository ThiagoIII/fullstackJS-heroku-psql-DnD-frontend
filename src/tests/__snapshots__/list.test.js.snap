// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h3>
    Quests
  </h3>,
  <ul>
    <li>
      quest1
    </li>
    <li>
      this is the first quest
    </li>
    <li>
      quest2
    </li>
    <li>
      this is the second quest
    </li>
    <li>
      quest3
    </li>
    <li>
      this is the third quest
    </li>
  </ul>,
]
`;

exports[`renders correctly: 
    Array [
      <h3>
        Quests
      </h3>,
      <ul>
        <li>
          quest1
        </li>
        <li>
          this is the first quest
        </li>
        <li>
          quest2
        </li>
        <li>
          this is the second quest
        </li>
        <li>
          quest3
        </li>
        <li>
          this is the third quest
        </li>
      </ul>,
    ]
   1`] = `
Array [
  <h3>
    Quests
  </h3>,
  <ul>
    <li>
      quest1
    </li>
    <li>
      this is the first quest
    </li>
    <li>
      quest2
    </li>
    <li>
      this is the second quest
    </li>
    <li>
      quest3
    </li>
    <li>
      this is the third quest
    </li>
  </ul>,
]
`;
